package/MDAnalysis/analysis/density.py

```diff
@@ -497,12 +495,70 @@ def __repr__(self):
             grid_type = 'histogram'
         return '<Density ' + grid_type + ' with ' + str(self.grid.shape) + ' bins>'
 
+
+def _set_user_grid(gridcenter, xdim, ydim, zdim, smin, smax):
+    """Helper function to set the grid dimensions to user defined values
+
+    Parameters
+    ----------
+    gridcenter : numpy ndarray, float32
+            3 element ndarray containing the x, y and z coordinates of the grid
+            box center
+    xdim : float
+            Box edge length in the x dimension
+    ydim : float
+            Box edge length in the y dimension
+    zdim : float
+            Box edge length in the y dimension
+    smin : numpy ndarray, float32
+            Minimum x,y,z coordinates for the input selection
+    smax : numpy ndarray, float32
+            Maximum x,y,z coordinates for the input selection
+
+    Returns
+    -------
+    umin : numpy ndarray, float32
+            Minimum x,y,z coordinates of the user defined grid
+    umax : numpy ndarray, float32
+            Maximum x,y,z coordinates of the user defined grid
+    """
+    # Check user inputs
+    if (len(gridcenter) != 3) or not isinstance(gridcenter, np.ndarray):
+        raise ValueError('gridcenter is improperly defined')
+    for entry in (xdim, ydim, zdim):
+        if not isinstance(entry, (float, int)):
+            raise ValueError("x/y/z dimensions are improperly defined")
+
+    # Copy gridcenter value to umin and umax
+    umin = np.copy(gridcenter)
+    umax = np.copy(gridcenter)
+
+    # Set min/max of each dimension by shifting by half the edge length
+    # x dimension
+    umin[0] -= xdim/2.0
```

Comment:
Using `xyzdim` from above, write more compact
```python
umin = gridcenter - xyzdim/2
umax = gridcenter + xyzdim/2
```

The copy lines above are not needed because we are making copies here.

The division will be true division because (1) the arrays are dtype float32 and (2) we have `from __future__ import division` at the top.

Reply:
As above should be addressed in 248e459